apps/desktop/src/components/StackDetails.svelte

```diff
@@ -12,43 +15,71 @@
 
 	const { stackId, projectId }: Props = $props();
 
+	const userSettings = getContextStoreBySymbol<Settings>(SETTINGS);
+	let resizeStackBranches = $state<HTMLElement>();
+	const stackBranchWidthKey = $derived('defaultStackBranchWidth_ ' + projectId);
+	let stackBranchWidth = $derived(persisted<number>(22.5, stackBranchWidthKey));
+
 	const stackService = getContext(StackService);
 	const result = $derived(stackService.getStackBranches(projectId, stackId));
 </script>
 
 <div class="stack">
-	<ReduxResult result={result.current}>
-		{#snippet children(result)}
-			{#if stackId && result.length > 0}
-				<div class="stack__branches">
-					{#each result as branch, i (branch.name)}
-						{@const first = i === 0}
-						{@const last = i === result.length - 1}
-						<Branch {branch} {first} {last} />
-					{/each}
-				</div>
-			{/if}
-		{/snippet}
-	</ReduxResult>
+	<div
+		class="stack__wrapper"
+		bind:this={resizeStackBranches}
+		style:width={$stackBranchWidth + 'rem'}
+	>
+		<Resizer
+			viewport={resizeStackBranches}
+			direction="right"
+			minWidth={22.5}
+			onWidth={(value) => {
+				$stackBranchWidth = value / (16 * $userSettings.zoom);
+			}}
+		/>
+		<div
+			class="stack__branches"
+			bind:this={resizeStackBranches}
+			style:width={$stackBranchWidth + 'rem'}
+		>
+			<ReduxResult result={result.current}>
+				{#snippet children(result)}
+					{#if stackId && result.length > 0}
+						{#each result as branch, i (branch.name)}
+							{@const first = i === 0}
+							{@const last = i === result.length - 1}
+							<Branch {branch} {first} {last} />
+						{/each}
+					{/if}
+				{/snippet}
+			</ReduxResult>
+		</div>
+	</div>
 
 	<div class="stack__branch-content">
 		<StackContentPlaceholder />
 	</div>
 </div>
 
 <style>
+	.stack__wrapper,
 	.stack {
```

Comment:
Can we look at this together? It feels weird that you have two containers nested, with the exact same css.

Reply:
WIP here: https://github.com/gitbutlerapp/gitbutler/pull/6242

Basically ready to go, but I dont want to rush it in now - I'll give it anotherh quick look over on Monday